app.py

```diff
@@ -64,26 +64,29 @@
     async with aiohttp.ClientSession(raise_for_status=True) as session:
```

Comment:
This is a general question and not related to the Unit tests as such...

Why not use `requests` for the POST action here like we do for other microservices?
Is `aiohttp` the recommended library when using `asyncio`?

Reply:
@AparnaKarve:

Function has to be a coroutine to be awaitable. You can read more about this concept of `async/await` in [Coroutines and Tasks Python docs](https://docs.python.org/3/library/asyncio-task.html).

Let's discuss what would happen if we convert `_retryable` to a coroutine (e.g. use the `async def` keyword):

1. A coroutine has to await something. In this case we want to await the network call.
2. An awaited network call means it's asynchronous. In other words the HTTP request should not block the IO/execution, until the response is retrieved. Therefore the thread is free to do other work until the response is received. At that point it would resume the work on the response.
3. To run the request in a non-blocking way, we need a library to support that. `requests` is not written to support that. And support is not considered. See the [requests docs for more info](http://docs.python-requests.org/en/master/user/advanced/#blocking-or-non-blocking).

Therefore we need to use a different async networking library in the place of `requests`. [`aiohttp`](https://github.com/aio-libs/aiohttp) is the most mature alternative, it has the biggest community and is part of a `aio-libs` collection, which is great in quality and support. Therefore I've choose this as replacement.

Other popular options would be [`requests-threads`](https://github.com/requests/requests-threads), [`grequests`](https://github.com/kennethreitz/grequests), [`requests-futures`](https://github.com/ross/requests-futures). All of them have smaller community, some are not even maintained anymore. `aiohttp` is the most future proof pick there I think.

Reply:
Thanks! I did read up a little bit yesterday on `aiohttp`, and the articles had good things to say about it.
I actually have a follow-up question, something that I encountered while working on the new Consumer service, so will post it in JIRA.

Reply:
Nice! This sounds very much like how we implement `Promise` in javascript.

Hopefully we can also use aiohttp and asyncio in other services as well.